lnwallet/channel.go

```diff
@@ -1973,8 +1991,11 @@ func NewBreachRetribution(chanState *channeldb.OpenChannel, stateNum uint64,
 
 	// With the commitment point generated, we can now generate the four
 	// keys we'll need to reconstruct the commitment state,
-	keyRing := deriveCommitmentKeys(commitmentPoint, false,
-		&chanState.LocalChanCfg, &chanState.RemoteChanCfg)
+	tweaklessCommit := chanState.ChanType == channeldb.SingleFunderTweakless
+	keyRing := deriveCommitmentKeys(
+		commitmentPoint, false, tweaklessCommit,
```

Comment:
To prevent all those `tweaklessCommit := chanState.ChanType == channeldb.SingleFunderTweakless` statements, the `ChanType` could be passed into `deriveCommitmentKeys`. Maybe a slightly different abstraction, but deriving keys for a specific channel type doesn't sound too bad.

Reply:
Ended up making a `IsTweakless` method to eliminate the code duplication.